src/playlist.rs

```diff
@@ -1144,6 +1225,253 @@ impl DateRange {
     }
 }
 
+// Implementing structs for LL-HLS
+#[derive(Debug, Default, PartialEq, Clone)]
+pub struct ServerControl {
+    pub can_skip_until: Option<f64>,
+    pub can_skip_dateranges: bool,
+    pub hold_back: Option<f64>,
+    pub part_hold_back: Option<f64>,
+    pub can_block_reload: bool,
+}
+
+impl ServerControl {
+    pub(crate) fn from_hashmap(
+        mut attrs: HashMap<String, QuotedOrUnquoted>,
+    ) -> Result<ServerControl, String> {
+        let can_skip_until = unquoted_string_parse!(attrs, "CAN-SKIP-UNTIL", |s: &str| s
+            .parse::<f64>()
+            .map_err(|err| format!("Failed to parse CAN-SKIP-UNTIL attribute: {}", err)));
+        let can_skip_dateranges = is_yes!(attrs, "CAN-SKIP-DATERANGES");
+        if can_skip_dateranges && can_skip_until.is_none() {
+            return Err(String::from(
+                "CAN-SKIP-DATERANGES attribute must be used with CAN-SKIP-UNTIL attribute",
+            ));
+        }
+        let hold_back = unquoted_string_parse!(attrs, "HOLD-BACK", |s: &str| s
+            .parse::<f64>()
+            .map_err(|err| format!("Failed to parse HOLD-BACK attribute: {}", err)));
+        let part_hold_back = unquoted_string_parse!(attrs, "PART-HOLD-BACK", |s: &str| s
+            .parse::<f64>()
+            .map_err(|err| format!("Failed to parse PART-HOLD-BACK attribute: {}", err)));
+        let can_block_reload = is_yes!(attrs, "CAN-BLOCK-RELOAD");
+
+        Ok(ServerControl {
+            can_skip_until,
+            can_skip_dateranges,
+            hold_back,
+            part_hold_back,
+            can_block_reload,
+        })
+    }
+
+    pub(crate) fn write_to<T: Write>(&self, w: &mut T) -> std::io::Result<()> {
+        write!(w, "#EXT-X-SERVER-CONTROL:")?;
+        write_some_float_attribute!(w, "CAN-SKIP-UNTIL", &self.can_skip_until)?;
+        if self.can_skip_dateranges {
+            write!(w, ",CAN-SKIP-DATERANGES=YES")?;
+        }
+        write_some_float_attribute!(w, ",HOLD-BACK", &self.hold_back)?;
+        write_some_float_attribute!(w, ",PART-HOLD-BACK", &self.part_hold_back)?;
+        if self.can_block_reload {
+            write!(w, ",CAN-BLOCK-RELOAD=YES")?;
+        }
+        writeln!(w)
+    }
+}
+
+#[derive(Debug, Default, PartialEq, Clone)]
+pub struct PartInf {
+    pub part_target: f64,
+}
+
+impl PartInf {
+    pub(crate) fn from_hashmap(
+        mut attrs: HashMap<String, QuotedOrUnquoted>,
+    ) -> Result<PartInf, String> {
+        let part_target = unquoted_string_parse!(attrs, "PART-TARGET", |s: &str| s
+            .parse::<f64>()
+            .map_err(|err| format!("Failed to parse PART-TARGET attribute: {}", err)))
+        .ok_or_else(|| String::from("EXT-X-PART-INF without mandatory PART-TARGET attribute"))?;
+
+        Ok(PartInf { part_target })
+    }
+
+    pub(crate) fn write_to<T: Write>(&self, w: &mut T) -> std::io::Result<()> {
+        match WRITE_OPT_FLOAT_PRECISION.load(Ordering::Relaxed) {
```

Comment:
Really liked the `write_some_float_attribute` you added, can something like this also be done for the non-option ones? I might look into that myself either way if u leave it like this

Reply:
I kind of hate this actually, but that may be my C++ brain talking that really wants a constexpr if or specialization here.

But that aside one could maybe wrap this in a macro, but I'm not entirely sure how much that would really help. The issue is that I'm not sure how you would support cases like [here](https://github.com/rutgersc/m3u8-rs/blob/e4004a142e60824ef434de57f1090e488be2a3ec/src/playlist.rs#L1347-L1362), where there are multiple replacements in the format string. The string itself also has to be different(`{:.*}` vs just `{}`), so overall it probably wouldn't really gain much vs. just spelling it out.

Reply:
Not familiar with the C++ stuff, but my guess even just looking at it would induce instant headache, same as with rust macros. 
Yeah its prob not worth it then.